File: domain/src/main/java/com/chronos/mobile/domain/EducationalTimetableHtmlParser.kt
package com.chronos.mobile.domain

import com.chronos.mobile.core.model.Timetable
import com.chronos.mobile.domain.result.AppResult

interface EducationalTimetableHtmlParser {
    fun parse(content: String): AppResult<Timetable?>
    fun parse(contentBytes: ByteArray): AppResult<Timetable?>
}

File: src/lib/domain/interfaces/educational-timetable-html-parser.ts
import type { Timetable } from '$lib/models/timetable';
import type { AppResult } from '../result';

export interface EducationalTimetableHtmlParser {
	parse(content: string): AppResult<Timetable | null>;
	parseBytes(contentBytes: Uint8Array): AppResult<Timetable | null>;
}
